src/utils/presenter.ts

```diff
@@ -29,6 +33,17 @@ export const postPresenterPage = async (req: Request, res: Response, next: NextF
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
 
     const session = req.session as Session;
+
+    if (isRemoveJourney(req)) {
+      const appData: ApplicationData = getApplicationData(session);
+      if (!appData[Transactionkey]) {
+        const transactionID = await postTransaction(req, session);
+        appData[Transactionkey] = transactionID;
+        appData[IsRemoveKey] = true;
```

Comment:
We've recently talked about moving this setting of the 'is_remove' field earlier in the journey but leaving it here for now.

Reply:
looking at update.filing.controller, it looks like we are missing a call to 
      setExtraData(req.session, appData);
to save the modified appData (with transaction ids) back to the session.

Reply:
Will unit tests be added before this move out of draft?

Reply:
Yep, that's my plan.

I'll add in that call to `setExtraData()` too.

Reply:
Unit tests and that missing call now added.